tests/api/GetMetadataApiCest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+use Page\MediaMetadata as MetadataApp;
+
+/**
+ * Class GetMetadataApiCest
+ */
+class GetMetadataApiCest {
+	private $apiUrl;
+	private $params = [
+	];
+	private $metadata = [
+
+	];
+
+	/**
+	 * @param ApiTester $I
+	 */
+	public function _before(ApiTester $I) {
+		$this->apiUrl = MetadataApp::$URL . 'api/metadata';
+	}
+
+	public function _after(ApiTester $I) {
+	}
+
+	public function getMetadataAsUser(\Step\Api\User $I) {
+		$I->am('an App');
+		$I->wantTo('retrieve the metadata stored in the database');
+
+		$data = $I->getFilesDataForFolder('');
+		$params = $this->params;
+
+		$id = $data['testimage.jpg']['id'];
+		$this->getMetadata($I, $params, $id);
+	}
+
+	private function getMetadata(\Step\Api\User $I, $params, $id) {
+		$I->getUserCredentialsAndUseHttpAuthentication();
+		$params['fileIDs'] = strval($id);
+		$I->sendGET($this->apiUrl, $params);
```

Comment:
The best thing to do to understand the difference would be to write down the URL you're using in the test and the one used in your route. You should immediately see the difference.
If it's still a mystery, then it probably means you don't know what a GET request looks like and you should try to find a general description on the types of requests the browser can send. It will be useful when implementing more features.

Gallery uses different types of routes, it depends on the variables which are sent.

Also, run the test in debug mode to look at what Codeception is sending. It might give you some clues.